# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/base/gcc/gcc32/sig_pthread_h.patch
# ROCK Linux is Copyright (C) 1998 - 2006 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---


Wow! It looks like there is a problem in the glibc header structure.
bits/sigthread.h (included by signal.h) needs pthread_t which is not
defined if we don't include pthread.h.

--- ./gcc/config/i386/linux.h.orig	2002-08-15 14:21:56.000000000 +0200
+++ ./gcc/config/i386/linux.h	2002-08-15 14:22:20.000000000 +0200
@@ -221,6 +221,7 @@
    state data appropriately.  See unwind-dw2.c for the structs.  */
 
 #ifdef IN_LIBGCC2
+#include <pthread.h>
 #include <signal.h>
 #include <sys/ucontext.h>
 #endif
